clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -293,23 +348,19 @@ bool CheckedRegionFinder::isUncheckedStruct(QualType Qt, std::set<std::string> &
 
 void CheckedRegionFinder::addUncheckedAnnotation(CompoundStmt *S, int Localwild) {
   auto Cur = S->getWrittenCheckedSpecifier();
-
   llvm::FoldingSetNodeID Id;
   S->Profile(Id, *Context, true);
-  auto Search = Seen.find(Id);
 
-  if (Search == Seen.end()) {
-    auto Loc = S->getBeginLoc();
-    bool IsChecked = !hasUncheckedParameters(S) &&
-                   Cur == CheckedScopeSpecifier::CSS_None && Localwild == 0;
 
-    Map[Id] = IsChecked ? IS_CHECKED : IS_UNCHECKED;
+  auto Loc = S->getBeginLoc();
+  bool IsChecked = !hasUncheckedParameters(S) &&
+    Cur == CheckedScopeSpecifier::CSS_None && Localwild == 0;
 
-    // Don't add _Unchecked to top level functions.
-    if ((!IsChecked && !isFunctionBody(S))) {
-      Writer.InsertTextBefore(Loc, "_Unchecked ");
-    }
-  }
+  Map[Id] = IsChecked ? IS_CHECKED : IS_UNCHECKED;
+
+  // Don't add _Unchecked to top level functions.
```

Comment:
Why not?

Reply:
Because functions default to being unchecked scope, so it would be unnecessary to add
`void foo(void) _Unchecked {`